dkist/dataset/loader.py

```diff
@@ -0,0 +1,73 @@
+import importlib.resources as importlib_resources
+from pathlib import Path
+from functools import singledispatch
+
+from jsonschema.exceptions import ValidationError
```

Comment:
we might as well take this opportunity to fix #276 and add jsonschema to the setup.cfg file.